data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisService.java

```diff
@@ -96,8 +95,11 @@ public KinesisService(final KinesisSourceConfig kinesisSourceConfig,
         this.dynamoDbClient = kinesisClientFactory.buildDynamoDBClient(kinesisLeaseConfig.getLeaseCoordinationTable().getAwsRegion());
         this.kinesisClient = kinesisClientFactory.buildKinesisAsyncClient(kinesisSourceConfig.getAwsAuthenticationConfig().getAwsRegion());
         this.cloudWatchClient = kinesisClientFactory.buildCloudWatchAsyncClient(kinesisLeaseConfig.getLeaseCoordinationTable().getAwsRegion());
-        this.pipelineName = pipelineDescription.getPipelineName();
-        this.applicationName = pipelineName;
+        if (kinesisLeaseConfig.getPipelineIdentifier().isEmpty()) {
```

Comment:
Please also check for `null`.

Reply:
Thanks @dlvenable . I have made the changes.